pharus/component_interface.py

```diff
@@ -74,19 +73,39 @@ def __init__(self, name, component_config, static_config, jwt_payload: dict):
         else:
             self.dj_restriction = lambda: dict()
 
-        self.vm_list = [
-            dj.VirtualModule(
-                s,
-                s,
-                connection=dj.conn(
-                    host=jwt_payload["databaseAddress"],
-                    user=jwt_payload["username"],
-                    password=jwt_payload["password"],
-                    reset=True,
-                ),
-            )
-            for s in inspect.getfullargspec(self.dj_query).args
-        ]
+        if "dj_query" in component_config:
+            exec(component_config["dj_query"], globals(), lcls)
+            self.dj_query = lcls["dj_query"]
+            self.vm_list = [
```

Comment:
This should be named `self.tables` to be more clear on what it represents. Also from discussion, we'll need to add a `connection` attribute to be accessible for transactions.